webapp/channels/src/components/user_settings/notifications/user_settings_notifications.tsx

```diff
@@ -79,86 +84,90 @@ function getNotificationsStateFromProps(props: Props): State {
-            pushActivity = user.notify_props.push;
+        if (props.user.notify_props.push) {
+            pushActivity = props.user.notify_props.push;
         }
-        if (user.notify_props.push_status) {
-            pushStatus = user.notify_props.push_status;
+        if (props.user.notify_props.push_status) {
+            pushStatus = props.user.notify_props.push_status;
         }
 
-        if (user.notify_props.auto_responder_active) {
-            autoResponderActive = user.notify_props.auto_responder_active === 'true';
+        if (props.user.notify_props.auto_responder_active) {
+            autoResponderActive = props.user.notify_props.auto_responder_active === 'true';
         }
 
-        if (user.notify_props.auto_responder_message) {
-            autoResponderMessage = user.notify_props.auto_responder_message;
+        if (props.user.notify_props.auto_responder_message) {
+            autoResponderMessage = props.user.notify_props.auto_responder_message;
         }
     }
 
     let usernameKey = false;
-    let customKeys = '';
     let firstNameKey = false;
     let channelKey = false;
-
-    if (user.notify_props) {
-        if (user.notify_props.mention_keys) {
-            const keys = user.notify_props.mention_keys.split(',');
-
-            if (keys.indexOf(user.username) === -1) {
-                usernameKey = false;
-            } else {
-                usernameKey = true;
-                keys.splice(keys.indexOf(user.username), 1);
-                if (keys.indexOf(`@${user.username}`) !== -1) {
-                    keys.splice(keys.indexOf(`@${user.username}`), 1);
+    let isCustomKeysWithNotificationInputChecked = false;
+    const customKeysWithNotification: MultiInputValue[] = [];
+
+    if (props.user.notify_props) {
+        if (props.user.notify_props.mention_keys) {
+            const mentionKeys = props.user.notify_props.mention_keys.split(',').filter((key) => key.length > 0);
+            mentionKeys.forEach((mentionKey) => {
+                // Remove username(s) from list of keys
+                if (mentionKey !== props.user.username && mentionKey !== `@${props.user.username}`) {
+                    customKeysWithNotification.push({
+                        label: mentionKey,
+                        value: mentionKey,
+                    });
                 }
-            }
+            });
+
+            // Check if username is in list of keys, if so, set the checkbox to true
+            usernameKey = mentionKeys.includes(props.user.username);
 
-            customKeys = keys.join(',');
+            // Check if there are any keys in the list, if so, set the checkbox of custom keys to true
+            isCustomKeysWithNotificationInputChecked = customKeysWithNotification.length > 0;
         }
 
-        if (user.notify_props.first_name) {
-            firstNameKey = user.notify_props.first_name === 'true';
+        if (props.user.notify_props.first_name) {
```

Comment:
I don't think the `if` are needed (nor the next one). If it is not set, it is undefined, and undefined is also distinct from 'true'.

Reply:
yes correct removed it